packages/dds/map/src/directory.ts

```diff
@@ -1475,8 +1522,13 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
      * @param op - The message if from a remote delete, or null if from a local delete
      */
     private deleteSubDirectoryCore(subdirName: string, local: boolean) {
+        const previousValue = this.getSubDirectory(subdirName);
         // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
         // Might want to consider cleaning out the structure more exhaustively though?
-        return this._subdirectories.delete(subdirName);
+        const successfullyRemoved = this._subdirectories.delete(subdirName);
+        if (previousValue !== undefined) {
+            (previousValue as SubDirectory).emit("dispose", local, previousValue);
```

Comment:
This is not great pattern to raise events on other objects. It's implicit flow that is hard to track.
I'd suggest to add methods on internal interfaces and calling these methods instead.

Reply:
Made changes.